compiler/plugins/target/AMD-AIE/iree-amd-aie/IR/AMDAIEDmaOpInterface.td

```diff
@@ -0,0 +1,94 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef IREE_AMDAIE_DIALECT_DMAOPINTERFACE
+#define IREE_AMDAIE_DIALECT_DMAOPINTERFACE
+
+include "mlir/IR/OpBase.td"
+
+//===----------------------------------------------------------------------===//
+// Defines the interface for dma-like operations.
+//===----------------------------------------------------------------------===//
+
+def DoublyStridedOpInterface : OpInterface<"DoublyStridedInterface"> {
+  let description = [{
+    An operation with strided access pattern on both source and target.
+
+    The strided access on both source and target is defined by `offsets`, `sizes`
+    and `strides`, all of which are expected to have the same number of dimensions.
+      - `offsets`: defines the base offset in every dimension from which the strided
+        access pattern starts.
+      - `sizes`: defines the number of accesses in every dimension for a corresponding
+        stride.
+      - `strides`: defines the strides or number of elements to be skipped to go to the
+        next access in every dimension.
+  }];
+  let cppNamespace = "mlir::iree_compiler::AMDAIE";
+
+  let methods = [
+    InterfaceMethod<
+      /*desc=*/"Returns ...",
+      /*retTy=*/"::mlir::ValueRange",
```

Comment:
While we are adding these maybe just do these the same way `subview` does these things (https://github.com/llvm/llvm-project/blob/2996cb7970fb536d7db1088d571dc8d2f9ad35fa/mlir/include/mlir/Dialect/MemRef/IR/MemRefOps.td#L1833) . This makes it much easier to handle constant propagation etc. especially if you want to have static sizes.

Reply:
This should be addressed in this new PR: https://github.com/nod-ai/iree-amd-aie/pull/271.